### PR TITLE
Emoji support

### PR DESCRIPTION
**Pull request by saralein**
Adds a selection of emojis for users to pick as tokens.

### CODE REVIEW

**Review by zdavy — APPROVED** [review]

File: lib/scripts/game_script.rb [hunk 1]
```diff
@@ -16,16 +16,29 @@ def initialize(colorizer)
...
+  def create_token_prompt(emojis, player)
+    output = "#{@token_selection}#{player}.\n\n     "
+
+    emojis.each_with_index do |code, code_index|
+      output += "#{colorizer.green("[#{code_index + 1}]".ljust(5, ' '))} #{code}"
+      if (code_index != emojis.length - 1)
+        output+= ' , '
+      end
+
+      if ((code_index + 1) != emojis.length && (code_index + 1) % 10 == 0)
```
Comment (damonkelley) [inline lib/scripts/game_script.rb]: what does this do?

File: lib/scripts/game_script.rb [hunk 2]
```diff
@@ -16,16 +16,29 @@ def initialize(colorizer)
...
-  def create_token_prompt(player)
-    "#{@token_selection}#{player}: "
+  def create_token_prompt(emojis, player)
+    output = "#{@token_selection}#{player}.\n\n     "
+
+    emojis.each_with_index do |code, code_index|
+      output += "#{colorizer.green("[#{code_index + 1}]".ljust(5, ' '))} #{code}"
+      if (code_index != emojis.length - 1)
+        output+= ' , '
+      end
```
Comment (damonkelley) [inline lib/scripts/game_script.rb]: Can this be simpler? Maybe you can use `map` and `join`?

File: spec/game_setup_spec.rb [hunk 3]
```diff
@@ -42,19 +42,25 @@
...
+      user_interface.set_input(['a', '150'])
+      expect(game_setup.get_token(config, setup_validator, 'Player 1')).to eql("\u{1f4a9}")
     end
 
     it 'tries again if token is taken' do
-      user_interface.set_input(['a', 'b'])
-      game_setup.instance_variable_set(:@token1, 'a')
-      expect(game_setup.get_token(config, setup_validator, 'Player 1')).to eql('b')
+      user_interface.set_input(['150', '149'])
+      game_setup.instance_variable_set(:@token1, '150')
```
Comment (damonkelley) [inline spec/game_setup_spec.rb]: 😢

File: lib/scripts/game_script.rb (hunk 2 above)
Comment (saralein) [reply lib/scripts/game_script.rb]: I divide this up into [multiple functions](https://github.com/saralein/tic-tac-toe_ruby/pull/22/files#diff-818a75ea9e19d71dfb4d1f5419d9d436R27) which I hope will make it easier to follow.

File: lib/scripts/game_script.rb (hunk 1 above)
Comment (saralein) [reply lib/scripts/game_script.rb]: Pulled this into a [function](https://github.com/saralein/tic-tac-toe_ruby/pull/22/files#diff-818a75ea9e19d71dfb4d1f5419d9d436R43) which I hope will better indicate what it's doing.

File: spec/game_setup_spec.rb (hunk 3 above)
Comment (saralein) [reply spec/game_setup_spec.rb]: I think this test is a little outdated with some of the other changes.  I've added a [method](https://github.com/saralein/tic-tac-toe_ruby/pull/22/files#diff-b1e3c91fecf2e775c58f0ba6d8acb4ecR77) to remove token 1 from emojis and included a [test](https://github.com/saralein/tic-tac-toe_ruby/pull/22/files#diff-affc662720a70899519e4bd6f5bd03e0R61) to make sure this happens.  This way, they will never have the option to pick token 1, only to pick something not in range.

I'm pretty sure you were saddened by instance_variable set.  I've updated another game setup test to [remove that](https://github.com/saralein/tic-tac-toe_ruby/pull/22/files#diff-affc662720a70899519e4bd6f5bd03e0L94).

**Review by damonkelley — APPROVED** [review]
🎉

**Pull request merged**